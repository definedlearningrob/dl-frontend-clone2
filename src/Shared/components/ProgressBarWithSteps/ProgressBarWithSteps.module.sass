@import '@shared/stylesheets/config.sass'

$progress-bar-height: $m-x
$progress-bar-height-xxxl: $m-sm

.stepsWrapper
  +flex-container(row)
  height: $progress-bar-height
  border-radius: $m-xs
  overflow: hidden
  gap: $m-xxxs

  +media-breakpoint-up(xxxl)
    height: $progress-bar-height-xxxl

.step
  background-color: $color-neutral-300
  flex: 1

.required
  background-color: $color-primary-300

.completed
  background-color: $color-primary-500
