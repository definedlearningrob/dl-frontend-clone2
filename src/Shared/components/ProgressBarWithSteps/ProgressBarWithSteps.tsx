import cx from 'classnames';

import styles from './ProgressBarWithSteps.module.sass';

type Props = {
  totalSteps: number;
  currentStep: number;
  minSteps?: number;
  className?: string;
};

export const ProgressBarWithSteps = ({
  currentStep,
  minSteps = 0,
  totalSteps,
  className,
}: Props) => {
  const stepsWrapperClassName = cx(styles.stepsWrapper, className);

  const steps = Array.from({ length: totalSteps }, (element, index) => {
    const stepClassName = cx(styles.step, {
      [styles.required]: minSteps > index,
      [styles.completed]: currentStep > index,
    });

    return (
      <div key={`step-${index}`} className={stepClassName} data-testid={`progress-step-${index}`} />
    );
  });

  return (
    <div
      aria-valuemax={totalSteps}
      aria-valuemin={minSteps}
      aria-valuenow={currentStep}
      className={stepsWrapperClassName}
      role='progressbar'>
      {steps}
    </div>
  );
};
