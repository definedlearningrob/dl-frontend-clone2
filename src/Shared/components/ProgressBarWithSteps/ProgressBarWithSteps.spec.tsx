import { render, screen } from '@testing-library/react';

import { ProgressBarWithSteps } from '@shared/components/ProgressBarWithSteps/ProgressBarWithSteps';

describe('ProgressBarWithSteps', () => {
  it('should render correctly without minSteps prop', () => {
    const { container } = render(
      <ProgressBarWithSteps className='additional-class-name' currentStep={1} totalSteps={5} />
    );

    const steps = screen.getAllByTestId(/step-/);
    expect(steps).toHaveLength(5);
    expect(steps[0]).toHaveClass('completed');
    expect(steps[0]).not.toHaveClass('required');

    expect(container).toMatchSnapshot();
  });

  it('should render correctly with minSteps prop', () => {
    const { container } = render(
      <ProgressBarWithSteps
        className='additional-class-name'
        currentStep={1}
        minSteps={2}
        totalSteps={5}
      />
    );

    const steps = screen.getAllByTestId(/step-/);
    expect(steps).toHaveLength(5);
    expect(steps[0]).toHaveClass('completed');
    expect(steps[0]).toHaveClass('required');
    expect(steps[1]).toHaveClass('required');

    expect(container).toMatchSnapshot();
  });
});
