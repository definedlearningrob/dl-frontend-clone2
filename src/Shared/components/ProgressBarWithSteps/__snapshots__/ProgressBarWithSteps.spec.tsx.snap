// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBarWithSteps should render correctly with minSteps prop 1`] = `
<div>
  <div
    aria-valuemax="5"
    aria-valuemin="2"
    aria-valuenow="1"
    class="stepsWrapper additional-class-name"
    role="progressbar"
  >
    <div
      class="step required completed"
      data-testid="progress-step-0"
    />
    <div
      class="step required"
      data-testid="progress-step-1"
    />
    <div
      class="step"
      data-testid="progress-step-2"
    />
    <div
      class="step"
      data-testid="progress-step-3"
    />
    <div
      class="step"
      data-testid="progress-step-4"
    />
  </div>
</div>
`;

exports[`ProgressBarWithSteps should render correctly without minSteps prop 1`] = `
<div>
  <div
    aria-valuemax="5"
    aria-valuemin="0"
    aria-valuenow="1"
    class="stepsWrapper additional-class-name"
    role="progressbar"
  >
    <div
      class="step completed"
      data-testid="progress-step-0"
    />
    <div
      class="step"
      data-testid="progress-step-1"
    />
    <div
      class="step"
      data-testid="progress-step-2"
    />
    <div
      class="step"
      data-testid="progress-step-3"
    />
    <div
      class="step"
      data-testid="progress-step-4"
    />
  </div>
</div>
`;
